### PR TITLE
ci: add sync files

### PR DESCRIPTION
**Pull request by wep21**
Signed-off-by: wep21 <border_goldenmarket@yahoo.co.jp>

## PR Type

- CI

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

File: .github/sync-files.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - source: .github/workflows/spell-check.yaml
+    - source: .clang-format
+    - source: .markdown-link-check.json
+    - source: .markdownlint.yaml
+    - source: .pre-commit-config-optional.yaml
+    - source: .prettierignore
+    - source: .prettierrc.yaml
+    - source: .yamllint.yaml
+    - source: CPPLINT.cfg
+    - source: setup.cfg
```
Comment (kenji-miyake) [inline .github/sync-files.yaml:16]: Don't you have to add some build-related workflows?

File: .github/workflows/sync-files.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  sync-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate token
+        id: generate-token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
```
Comment (kenji-miyake) [inline .github/workflows/sync-files.yaml:17]: We have to set the key. I'll do it.

**Review by kenji-miyake — APPROVED** [review]

File: .github/sync-files.yaml (hunk 1 above)
Comment (wep21) [reply .github/sync-files.yaml:16]: update at https://github.com/tier4/ros2-project-template/pull/2/commits/c1cb583a00af7c06662225bbab09092b69ba8422

File: .github/workflows/sync-files.yaml (hunk 2 above)
Comment (kenji-miyake) [reply .github/workflows/sync-files.yaml:17]: Done.

File: .github/sync-files.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    - source: .pre-commit-config-optional.yaml
+    - source: .prettierignore
+    - source: .prettierrc
+    - source: .yamllint.yaml
+    - source: CPPLINT.cfg
+    - source: setup.cfg
+
+- repository: autowarefoundation/autoware_common
+  files:
+    - source: .github/workflows/automatic-rebase.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml]: I've added this to autowarefoundation/autoware, so let's copy from here.
https://github.com/autowarefoundation/autoware/blob/tier4/proposal/.github/workflows/automatic-rebase.yaml

File: .github/sync-files.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+- repository: autowarefoundation/autoware
+  files:
+    - source: .github/dependabot.yaml
+    - source: .github/workflows/pre-commit-optional.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml:5]:
```suggestion
    - source: .github/workflows/automatic-rebase.yaml
    - source: .github/workflows/pre-commit-optional.yaml
```

File: .github/sync-files.yaml (hunk 3 above)
Comment (kenji-miyake) [inline .github/sync-files.yaml]:
```suggestion
```

Comment (kenji-miyake) [reply .github/sync-files.yaml]: https://github.com/tier4/ros2-project-template/pull/2/files#r799262074
https://github.com/tier4/ros2-project-template/pull/2/files#r799262145

File: .github/sync-files.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    - source: .prettierrc
+    - source: .yamllint.yaml
+    - source: CPPLINT.cfg
+    - source: setup.cfg
+
+- repository: autowarefoundation/autoware_common
+  files:
+    - source: .github/workflows/automatic-rebase.yaml
+    - source: .github/workflows/build-and-test-arm.yaml
+    - source: .github/workflows/build-and-test-arm-pr.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml]: For ARM workflows, we need to set up self-hosted runners, but we haven't done that.

```suggestion
```

File: .github/sync-files.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    - source: .github/workflows/build-and-test.yaml
+    - source: .github/workflows/build-and-test-pr.yaml
+    - source: .clang-format
```
Comment (kenji-miyake) [inline .github/sync-files.yaml]: This was my mistake.
https://github.com/autowarefoundation/autoware.universe/pull/311#discussion_r795081291

```suggestion
    - source: .github/workflows/pre-commit.yaml
```

File: .github/sync-files.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    - source: .github/workflows/automatic-rebase.yaml
+    - source: .github/workflows/pre-commit-optional.yaml
+    - source: .github/workflows/semantic-pull-request.yaml
+    - source: .github/workflows/spell-check.yaml
+    - source: .clang-format
+    - source: .markdown-link-check.json
+    - source: .markdownlint.yaml
+    - source: .pre-commit-config-optional.yaml
+    - source: .prettierignore
+    - source: .prettierrc
```
Comment (kenji-miyake) [inline .github/sync-files.yaml]: This is also my mistake. I'm sorry.

```suggestion
    - source: .prettierrc.yaml
```

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**